### PR TITLE
[docs] Update community pipeline docs

### PR DESCRIPTION
**Pull request by stevhliu**
This PR updates the community pipeline docs:

- removes the [Adding custom pipelines to the Hub](https://huggingface.co/docs/diffusers/main/en/using-diffusers/custom_pipeline_overview#adding-custom-pipelines-to-the-hub) section because that is covered in [How to contribute a community pipeline](https://huggingface.co/docs/diffusers/main/en/using-diffusers/contribute_pipeline); keeps this doc focused on loading community pipelines
- gives a little more structure to [How to contribute a community pipeline](https://huggingface.co/docs/diffusers/main/en/using-diffusers/contribute_pipeline) and uses `diff` blocks for the code examples so users can more easily see which lines of code were added
- minor clean up

To do:
- [x] explain the difference between official GitHub community pipelines and community pipelines shared to the Hugging Face Hub

### CODE REVIEW

File: docs/source/en/using-diffusers/contribute_pipeline.mdx [hunk 1]
```diff
@@ -85,71 +91,72 @@ pipeline = UnetSchedulerOneForwardPipeline(unet=unet, scheduler=scheduler)
...
+from diffusers import DiffusionPipeline
 
-        return scheduler_output
+pipe = DiffusionPipeline.from_pretrained("google/ddpm-cifar10-32", custom_pipeline="one_step_unet")
+pipe()
 ```
 
-Our amazing pipeline got merged here: [#840](https://github.com/huggingface/diffusers/pull/840).
-Now everybody that has `diffusers >= 0.4.0` installed can use our pipeline magically 🪄 as follows:
+Another way to share your community pipeline is to upload the `one_step_unet.py` file directly to your preferred [model repository](https://huggingface.co/docs/hub/models-uploading) on the Hub. Instead of specifying the `one_step_unet.py` file, pass the model repository id to the `custom_pipeline` argument:
```
Comment (stevhliu) [inline docs/source/en/using-diffusers/contribute_pipeline.mdx:115]: What's the difference between having the pipeline on GitHub vs Hugging Face Hub? It seems like the GitHub ones are "official" and the Hub ones aren't. The only difference seems to be the sharing workflow (kind of like the [legacy datasets](https://huggingface.co/docs/datasets/share#datasets-on-github-legacy) from the Datasets library), so I think it'd be nice to clarify for the user here the pros and cons of each sharing workflow.

cc @patrickvonplaten @yiyixuxu @sayakpaul @pcuenca

Comment (pcuenca) [reply docs/source/en/using-diffusers/contribute_pipeline.mdx:115]: I think the only difference is that GitHub ones are reviewed and added to the repo, but users can still use others from the Hub without going through this process.

Comment (stevhliu) [reply docs/source/en/using-diffusers/contribute_pipeline.mdx:115]: Is there a recommended way to share the pipeline? I think this can potentially be confusing for users, for example, are the Hub community pipelines not as "official" or "legit" as the GitHub community ones? I made a table comparing the two:

|                | GitHub community pipeline                                             | HF Hub community pipeline                                       |
|----------------|-----------------------------------------------------------------------|-----------------------------------------------------------------|
| visibility     | included in official Diffusers examples                               | relies on your own usage and promotion                          |
| review process | open a Pull Request on GitHub and get it merged by the Diffusers team | upload directly to a Hub repository without any review (faster) |
| usability      | same                                                                  | same                                                            |

Comment (patrickvonplaten) [reply docs/source/en/using-diffusers/contribute_pipeline.mdx:115]: @stevhliu I think you table is great and that's exactly the trade-off! 
Would be super nice to include it

File: docs/source/en/using-diffusers/contribute_pipeline.mdx [hunk 2]
```diff
@@ -43,39 +34,54 @@ import torch
...
+          self.register_modules(unet=unet, scheduler=scheduler)
 
-    def __call__(self):
-        image = torch.randn(
-            (1, self.unet.in_channels, self.unet.sample_size, self.unet.sample_size),
-        )
-        timestep = 1
++     def __call__(self):
++         image = torch.randn(
++             (1, self.unet.in_channels, self.unet.sample_size, self.unet.sample_size),
```
[the remark this answers is not in the document]
Comment (patrickvonplaten) [reply docs/source/en/using-diffusers/contribute_pipeline.mdx:71]: Just changed that API a bit

**Review by patrickvonplaten — APPROVED** [review]
PR looks very nice to me!
